packages/openapi-to-graphql/src/resolver_builder.ts

```diff
@@ -559,92 +574,60 @@ export function getResolver<TSource, TContext, TArgs>({
       } else if (
         operation.payloadContentType === 'application/x-www-form-urlencoded'
       ) {
-        rawPayload = formurlencoded(
+        payload = formurlencoded(
           Oas3Tools.desanitizeObjectKeys(args[sanePayloadName], data.saneMap)
         )
       } else {
         // Payload is not an object
-        rawPayload = args[sanePayloadName]
+        payload = args[sanePayloadName]
       }
-      options.body = rawPayload
-      resolveData.usedPayload = rawPayload
     }
 
-    /**
-     * Pass on OpenAPI-to-GraphQL options
-     */
-    if (typeof data.options === 'object') {
-      // Headers:
-      if (typeof data.options.headers === 'object') {
-        Object.assign(options.headers, data.options.headers)
-      } else if (typeof data.options.headers === 'function') {
-        const headers = data.options.headers(method, path, title, {
-          source,
-          args,
-          context,
-          info
-        })
-
-        if (typeof headers === 'object') {
-          Object.assign(options.headers, headers)
-        }
-      }
-
-      // Query string:
-      if (typeof data.options.qs === 'object') {
-        Object.assign(options.qs, data.options.qs)
-      }
+    let options: NodeRequest.OptionsWithUrl = {
+      method: operation.method,
+      url,
+      headers,
+      qs,
+      body: payload,
+      jar: cookieJar // For authentication cookies
     }
 
-    // Get authentication headers and query parameters
-    if (
-      source &&
-      typeof source === 'object' &&
-      typeof source[OPENAPI_TO_GRAPHQL] === 'object'
-    ) {
-      const { authHeaders, authQs, authCookie } = getAuthOptions(
-        operation,
-        source[OPENAPI_TO_GRAPHQL],
-        data
-      )
-
-      // ...and pass them to the options
-      Object.assign(options.headers, authHeaders)
-      Object.assign(options.qs, authQs)
-
-      // Add authentication cookie if created
-      if (authCookie !== null) {
-        const j = NodeRequest.jar()
-        j.setCookie(authCookie, options.url)
-        options.jar = j
-      }
-    }
+    // The options `requestOptions` will finalize the options
+    if (requestOptions) {
+      /**
+       * If requestOptions is a function, the run the function, otherwise,
+       * use the original value
+       */
+      const requestOptionsValue =
+        typeof requestOptions === 'function'
+          ? requestOptions(
+              method,
+              path,
+              title,
+              { source, args, context, info },
+              options
+            )
+          : requestOptions
 
-    // Extract OAuth token from context (if available)
-    if (data.options.sendOAuthTokenInQuery) {
-      const oauthQueryObj = createOAuthQS(data, context)
-      Object.assign(options.qs, oauthQueryObj)
-    } else {
-      const oauthHeader = createOAuthHeader(data, context)
-      Object.assign(options.headers, oauthHeader)
+      options = merge(options, requestOptionsValue)
```

Comment:
Notice that we merge the original options with `requestOptionsValue`. If we were to implement a `requestOptionsStrict` option like I mentioned previously, if would do something here.